### PR TITLE
doc: refine CLI help doc

### PR DESCRIPTION
**Pull request by FeynmanZhou**
fix https://github.com/oras-project/oras/issues/522
Signed-off-by: Feynman Zhou <feynmanzhou@microsoft.com>

### CODE REVIEW

File: cmd/oras/attach.go [hunk 1]
```diff
@@ -62,6 +62,9 @@ Example - Attach an artifact with manifest annotations
 Example - Attach a file and add manifest annotations
   oras attach --artifact-type doc/example --annotation "key=val" localhost:5000/hello:latest hi.txt
 
+Example - Attach a file and export the pushed manifest to a specified path
```
Comment (qweeah) [inline cmd/oras/attach.go:65]:
```suggestion
Example - Attach file 'hi.txt' and export the pushed manifest to 'manifest.json'
```

File: cmd/oras/blob/fetch.go [hunk 2]
```diff
@@ -79,7 +79,7 @@ Example - Fetch the blob, save it to a local file and stdout the descriptor:
 		},
 	}
 
-	cmd.Flags().StringVarP(&opts.outputPath, "output", "o", "", "output file path")
+	cmd.Flags().StringVarP(&opts.outputPath, "output", "o", "", "write it to standard output")
```
Comment (qweeah) [inline cmd/oras/blob/fetch.go:82]:
```suggestion
	cmd.Flags().StringVarP(&opts.outputPath, "output", "o", "", "output file `path`, use - for stdout")
```

File: cmd/oras/login.go [hunk 3]
```diff
@@ -41,22 +41,22 @@ func loginCmd() *cobra.Command {
 		Short: "Log in to a remote registry",
 		Long: `Log in to a remote registry
 
-Example - Login with username and password from command line:
+Example - Log in with username and password from user input:
```
Comment (qweeah) [inline cmd/oras/login.go:44]:
```suggestion
Example - Log in with username and password from command line input:
```

File: cmd/oras/login.go [hunk 4]
```diff
@@ -41,22 +41,22 @@ func loginCmd() *cobra.Command {
...
-Example - Login with identity token from stdin:
+Example - Log in with identity token from stdin:
   oras login --password-stdin localhost:5000
 
-Example - Login with username and password by prompt:
+Example - Log in with username and password in an interactive terminal:
   oras login localhost:5000
 
-Example - Login with insecure registry from command line:
+Example - Log in to an insecure registry from user input:
```
Comment (qweeah) [inline cmd/oras/login.go:59]:
```suggestion
Example - Log in with username and password in an interactive terminal and no TLS check:
```

File: cmd/oras/manifest/fetch.go [hunk 5]
```diff
@@ -80,7 +80,7 @@ Example - Fetch manifest with prettified json result:
 	}
 
 	cmd.Flags().StringSliceVarP(&opts.mediaTypes, "media-type", "", nil, "accepted media types")
-	cmd.Flags().StringVarP(&opts.outputPath, "output", "o", "", "output file path")
+	cmd.Flags().StringVarP(&opts.outputPath, "output", "o", "", "write it to standard output")
```
Comment (qweeah) [inline cmd/oras/manifest/fetch.go:83]:
```suggestion
	cmd.Flags().StringVarP(&opts.outputPath, "output", "o", "", "file `path` to write the fetched manifest to, use - for stdout")
```

File: cmd/oras/manifest/fetch_config.go [hunk 6]
```diff
@@ -84,7 +84,7 @@ Example - Fetch and print the prettified descriptor of the config:
 		},
 	}
 
-	cmd.Flags().StringVarP(&opts.outputPath, "output", "o", "", "output file path")
+	cmd.Flags().StringVarP(&opts.outputPath, "output", "o", "", "write it to standard output")
```
Comment (qweeah) [inline cmd/oras/manifest/fetch_config.go:87]:
```suggestion
	cmd.Flags().StringVarP(&opts.outputPath, "output", "o", "", "file `path` to write the fetched config to, use - for stdout")
```

File: cmd/oras/tag/tag.go [hunk 7]
```diff
@@ -50,7 +50,7 @@ Example - Tag the manifest with digest sha256:9463e0d192846bc994279417b501146067
 Example - Tag the manifest 'v1.0.1' in 'localhost:5000/hello' to 'v1.0.2', 'latest'
   oras tag localhost:5000/hello:v1.0.1 v1.0.2 latest
 
-Example - Tag the manifest 'v1.0.1' in 'localhost:5000/hello' to 'v1.0.2' 'latest' with concurrency level tuned:
+Example - Tag the manifest 'v1.0.1' in 'localhost:5000/hello' to 'v1.0.2', 'latest' with concurrency level tuned:
```
Comment (qweeah) [inline cmd/oras/tag/tag.go:53]:
```suggestion
Example - Tag the manifest 'v1.0.1' in 'localhost:5000/hello' to 'v1.0.1', 'v1.0.2', 'latest' with concurrency level tuned:
```

**Review by qweeah — CHANGES_REQUESTED** [review]
Thanks a lot for revising the docs!

**Comment by qweeah** [comment]
@FeynmanZhou Can you rebase the PR to the latest main branch?

File: cmd/oras/tag/tag.go [hunk 8]
```diff
@@ -50,7 +50,7 @@ Example - Tag the manifest with digest sha256:9463e0d192846bc994279417b501146067
 Example - Tag the manifest 'v1.0.1' in 'localhost:5000/hello' to 'v1.0.2', 'latest'
   oras tag localhost:5000/hello:v1.0.1 v1.0.2 latest
 
-Example - Tag the manifest 'v1.0.1' in 'localhost:5000/hello' to 'v1.0.2' 'latest' with concurrency level tuned:
+Example - Tag the manifest 'v1.0.1' in 'localhost:5000/hello' to 'v1.0.1', 'v1.0.2', 'latest' with concurrency level tuned:
```
Comment (qweeah) [inline cmd/oras/tag/tag.go:53]: Sorry just realize that I made a mistake, can you help revert the change

```suggestion
Example - Tag the manifest 'v1.0.1' in 'localhost:5000/hello' to 'v1.0.2', 'latest' with concurrency level tuned:
```

**Review by qweeah — CHANGES_REQUESTED** [review]
Looks good to me.

**Review by qweeah — APPROVED** [review]
Looks good to me.

File: cmd/oras/attach.go [hunk 9]
```diff
@@ -59,9 +59,12 @@ Example - Attach file 'hi.txt' and add annotations from file 'annotation.json'
 Example - Attach an artifact with manifest annotations
   oras attach --artifact-type doc/example --annotation "key1=val1" --annotation "key2=val2" localhost:5000/hello:latest
 
-Example - Attach a file and add manifest annotations
+Example - Attach file 'hi.txt' and add manifest annotations
   oras attach --artifact-type doc/example --annotation "key=val" localhost:5000/hello:latest hi.txt
 
+Example - Attach file 'hi.txt' and export the pushed manifest to 'manifest.json'
+  oras attach --artifact-type doc/example --export-manifest manifest.json localhost:5000/hello:latest hi.txt
+
```
Comment (shizhMSFT) [inline cmd/oras/attach.go:67]: Remove empty line.

File: cmd/oras/internal/option/remote.go [hunk 10]
```diff
@@ -74,7 +74,7 @@ func (opts *Remote) ApplyFlagsWithPrefix(fs *pflag.FlagSet, prefix, description
 	fs.StringVarP(&opts.CACertFilePath, flagPrefix+"ca-file", "", "", "server certificate authority file for the remote "+notePrefix+"registry")
 
 	if fs.Lookup("registry-config") == nil {
-		fs.StringArrayVarP(&opts.Configs, "registry-config", "", nil, "auth config path")
+		fs.StringArrayVarP(&opts.Configs, "registry-config", "", nil, "path of the authentication file")
```
Comment (shizhMSFT) [inline cmd/oras/internal/option/remote.go:77]: nit:
```suggestion
		fs.StringArrayVarP(&opts.Configs, "registry-config", "", nil, "`path` of the authentication file")
```

File: cmd/oras/login.go (hunk 3 above)
Comment (shizhMSFT) [inline cmd/oras/login.go:44]: nit:
```suggestion
Example - Log in with username and password from command line arguments:
```

File: cmd/oras/login.go [hunk 11]
```diff
@@ -41,22 +41,22 @@ func loginCmd() *cobra.Command {
...
-Example - Login with username and password from command line:
+Example - Log in with username and password from user input:
   oras login -u username -p password localhost:5000
 
-Example - Login with username and password from stdin:
+Example - Log in with username and password from stdin:
   oras login -u username --password-stdin localhost:5000
 
-Example - Login with identity token from command line:
+Example - Log in with identity token from user input:
```
Comment (shizhMSFT) [inline cmd/oras/login.go:50]:
```suggestion
Example - Log in with identity token from command line arguments:
```

File: cmd/oras/login.go (hunk 3 above)
Comment (FeynmanZhou) [reply cmd/oras/login.go:44]: user input => command line flags

File: cmd/oras/login.go (hunk 11 above)
Comment (FeynmanZhou) [reply cmd/oras/login.go:50]: Same as above

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**